src/measure_theory/integral/layercake.lean

```diff
@@ -0,0 +1,258 @@
+/-
+Copyright (c) 2022 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import measure_theory.integral.interval_integral
+import analysis.special_functions.integrals
+
+/-!
+# The layer cake formula, a.k.a. Cavalieri's principle
+
+In this file we prove the layer cake formula using Fubini's theorem and the two most common
```

Comment:
I don't know what the "layer cake formula" is (probably, I know the formula, but not under this name). Could you explain in this docstring what the formula says?

Reply:
I did not know a name for this either! The two names I use here were [pointed out to me by Heather and Patrick](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Portmanteau.20theorem/near/244488754). I would have called this something much more mundane, and I think I should in fact seek a mathlib style naming of the results, since the fancier terminology is not very standard.

I'd believe the main problem with this PR is that my proofs are far from elegant. But I should obviously also improve the docstrings as you point out. Sorry about not having put more effort into them before, I now realize it might have made this less awful to review.

Reply:
But although my proofs are clumsy, I think the result itself is worth adding when acceptable. The direct motivation for me was one implication of portmanteau theorem (which only needs `layercake_one`). Just today I learned that Jason is about to use essentially `layercake_p` in Doob's L^p maximal inequalities.

Jason had another proof of `layercake_p` in #14769, but the current PR seems to have a bit more more general assumptions, so to me it looks better to obtain the other one as a "special case" (strictly speaking, some glue is needed anyway, since so far I formulated things with `lintegral` and Jason with `integral`). Jason also kindly promised to take a look into improving my clumsy proof. He clearly has more formalization experience. And we noted having struggled with some similar issues in our independent proofs of essentially the same result.

Reply:
I renamed the results as [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Naming.20contest/near/286702246).

Since "layer cake" is not recognized by almost anyone, should the file name also be changed?